ee/app/livechat-enterprise/server/api/units.ts

```diff
@@ -0,0 +1,93 @@
+import { API } from '../../../../../app/api/server';
+import { deprecationWarning } from '../../../../../app/api/server/helpers/deprecationWarning';
+import { findUnits, findUnitById, findUnitMonitors } from './lib/units';
+import { LivechatEnterprise } from '../lib/LivechatEnterprise';
+import { IOmnichannelBusinessUnit } from '../../../../../definition/IOmnichannelBusinessUnit';
+
+API.v1.addRoute('livechat/units.list', { authRequired: true }, {
+	async get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort } = this.parseJsonQuery();
+		const { text } = this.queryParams;
+
+		const response = await findUnits({
+			userId: this.userId,
+			text,
+			pagination: {
+				offset,
+				count,
+				sort,
+			},
+		});
+
+		return API.v1.success(deprecationWarning({ response, endpoint: 'livechat/units.list' }));
+	},
+});
+
+API.v1.addRoute('livechat/units.getOne', { authRequired: true }, {
+	async get() {
+		const { id } = this.urlParams;
+		const { unit } = await findUnitById({
+			userId: this.userId,
+			unitId: id,
+		}) as { unit: IOmnichannelBusinessUnit };
+
+		return API.v1.success(deprecationWarning({ response: unit, endpoint: 'livechat/units.getOne' }));
+	},
+});
+
+API.v1.addRoute('livechat/unitMonitors.list', { authRequired: true }, {
+	async get() {
+		const { unitId } = this.queryParams;
+
+		return API.v1.success(await findUnitMonitors({
+			userId: this.userId,
+			unitId,
+		}));
+	},
+});
+
+API.v1.addRoute('livechat/units', { authRequired: true, permissionsRequired: ['manage-livechat-units'] }, {
+	async get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort } = this.parseJsonQuery();
+		const { text } = this.queryParams;
+
+		return API.v1.success(Promise.await(findUnits({
+			userId: this.userId,
+			text,
+			pagination: {
+				offset,
+				count,
+				sort,
+			},
+		})));
+	},
+	async post() {
+		const { unitData, unitMonitors, unitDepartments } = this.bodyParams?.();
+		return LivechatEnterprise.saveUnit(null, unitData, unitMonitors, unitDepartments);
+	},
+});
+
+API.v1.addRoute('livechat/units/:id', { authRequired: true, permissionsRequired: ['manage-livechat-units'] }, {
+	async get() {
+		const { id } = this.urlParams;
+		const { unit } = await findUnitById({
+			userId: this.userId,
+			unitId: id,
+		}) as { unit: IOmnichannelBusinessUnit };
+
+		return API.v1.success(unit);
+	},
+	async post() {
+		const { unitData, unitMonitors, unitDepartments } = this.bodyParams?.();
```

Comment:
What happens inside `LivechatEnterprise.saveUnit` if either `unitMonitors` or `unitDepartments` are not provided?

Reply:
It will throw an error. We have a check for that over there - see [here](https://github.com/RocketChat/Rocket.Chat/blob/d6a25f550ac8aa59f235f332e9648f29cfc3f421/ee/app/livechat-enterprise/server/lib/LivechatEnterprise.js#L75)

Reply:
Yup, all of those are required params 👀